drivers/wifi/simplelink_support.c

```diff
@@ -0,0 +1,616 @@
+#define DEVICE_ERROR	 ("See \"DEVICE ERRORS CODES\" in SimpleLink errors.h")
+#define WLAN_ERROR	 ("See \"WLAN ERRORS CODES\" in SimpleLink errors.h")
+#define NETAPP_ERROR	 ("See \"NETAPP ERRORS CODES\" in SimpleLink errors.h")
+
+#define CHANNEL_MASK_ALL	    (0x1FFF)
+#define RSSI_TH_MAX		    (-95)
+
+enum status_bits {
+	/* Network Processor is powered up */
+	STATUS_BIT_NWP_INIT = 0,
+	/* The device is connected to the AP */
+	STATUS_BIT_CONNECTION,
+	/* The device has leased IP to any connected client */
+	STATUS_BIT_IP_LEASED,
+	/* The device has acquired an IP */
+	STATUS_BIT_IP_ACQUIRED,
+	/* The device has acquired an IPv6 address */
+	STATUS_BIT_IPV6_ACQUIRED,
+};
+
+struct nwp_status {
+	/* Callback to notify net & wifi mgmt events from SL Event Handlers */
+	simplelink_wifi_cb_t cb;
+
+	/* Status Variables */
+	u32_t status;	/* The state of the NWP */
+	u32_t role;	/* The device's role (STA, P2P or AP) */
+
+	/* Scan results table: */
+	SlWlanNetworkEntry_t net_entries[CONFIG_WIFI_SIMPLELINK_SCAN_COUNT];
+};
+
+/* STA/AP mode state: shared with simplelink.c */
+struct sl_connect_state sl_conn;
+
+/* Network Coprocessor state, including role and connection state: */
+static struct nwp_status nwp;
+
+/* Configure the device to a default state, resetting previous parameters .*/
+static s32_t configure_simplelink(void)
+{
+	u8_t config_opt;
+	u8_t power;
+	s32_t retval = -1;
+	s32_t mode = -1;
+	u32_t if_bitmap = 0;
+	SlWlanScanParamCommand_t scan_default = { 0 };
+	SlWlanRxFilterOperationCommandBuff_t rx_filterid_mask = { { 0 } };
+
+	/* Turn on NWP */
+	mode = sl_Start(0, 0, 0);
+	ASSERT_ON_ERROR(mode, DEVICE_ERROR);
+
+	if (mode != ROLE_STA) {
+		/* Set NWP role as STA */
+		mode = sl_WlanSetMode(ROLE_STA);
+		ASSERT_ON_ERROR(mode, WLAN_ERROR);
+
+		/* For changes to take affect, we restart the NWP */
+		retval = sl_Stop(SL_STOP_TIMEOUT);
+		ASSERT_ON_ERROR(retval, DEVICE_ERROR);
+
+		mode = sl_Start(0, 0, 0);
+		ASSERT_ON_ERROR(mode, DEVICE_ERROR);
+	}
+
+	if (mode != ROLE_STA) {
+		SYS_LOG_ERR("Failed to configure NWP to default state");
+		return -1;
+	}
+
+	/* Remove Auto Connection Policy, to avoid multiple re-connect tries:
+	 * Note: this doesn't actually work.
+	 * NWP still issues multiple reconnects on a connection failure.
+	 */
+	retval = sl_WlanPolicySet(SL_WLAN_POLICY_CONNECTION,
+				  SL_WLAN_CONNECTION_POLICY(0, 0, 0, 0),
+				  NULL, 0);
+	ASSERT_ON_ERROR(retval, WLAN_ERROR);
+
+	/* Disable Auto Provisioning*/
+	retval = sl_WlanProvisioning(SL_WLAN_PROVISIONING_CMD_STOP, 0xFF, 0,
+				     NULL, 0x0);
+	ASSERT_ON_ERROR(retval, WLAN_ERROR);
+
+	/* Delete existing profiles */
+	retval = sl_WlanProfileDel(0xFF);
+	ASSERT_ON_ERROR(retval, WLAN_ERROR);
+
+	/* enable DHCP client */
+	retval = sl_NetCfgSet(SL_NETCFG_IPV4_STA_ADDR_MODE,
+			      SL_NETCFG_ADDR_DHCP, 0, 0);
+	ASSERT_ON_ERROR(retval, NETAPP_ERROR);
+
+	/* Disable ipv6 */
+	if_bitmap = !(SL_NETCFG_IF_IPV6_STA_LOCAL |
```

Comment:
Is there some bigger issue with IPv6, or do we disable this because this is just not tested properly?
I mean would it make sense to enable IPv6 if user has enabled CONFIG_NET_IPV6 , now we always disable it?